File: java-car-data/car-data-web-ui-tests/src/test/java/com/glicerial/samples/cardata/web/uitests/page/ChartsPage.java
package com.glicerial.samples.cardata.web.uitests.page;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;


public class ChartsPage {

    private WebDriver driver;

    @FindBy(id = "charts")
    private WebElement chartsDiv;

    public ChartsPage(WebDriver webDriver) {
        this.driver = webDriver;
        WebDriverWait wait = new WebDriverWait(driver, 5);
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.id("charts")));
        PageFactory.initElements(driver, this);
    }
}

File: python-selenium/pages/charts_page.py
from selenium.webdriver.support import expected_conditions
from selenium.webdriver.support.ui import WebDriverWait

from .locators import ChartsPageLocators
from .page import BasePage

class ChartsPage(BasePage):
    def __init__(self, web_driver):
        super().__init__(web_driver)

        self.wait = WebDriverWait(self.driver, 5)
        self.charts_div = self.wait.until(expected_conditions.visibility_of_element_located(ChartsPageLocators.CHARTS_DIV))
